front/lib/api/gens.ts

```diff
@@ -3,12 +3,18 @@ import { Op } from "sequelize";
 import { GensTemplateType } from "@app/types/gens";
 import { UserType, WorkspaceType } from "@app/types/user";
 
+import { Authenticator } from "../auth";
 import { GensTemplate } from "../models";
 
 export async function getGensTemplates(
-  owner: WorkspaceType,
+  auth: Authenticator,
   user: UserType
 ): Promise<GensTemplateType[]> {
+  const owner = auth.workspace();
```

Comment:
!owner seems to indicate an errror
Since [] is a valid return (e.g. if you just don't have templates) I would make a semantic distinction, e.g. by using null as done in getTemplate

Reply:
If auth has no workspace that means the user is not authenticated and returning an empty array is semantically OK + this is what we do in the rest of this directory. When we try to write we do error and that's unexpected as the caller of this function should make sure that the user is authenticated